Android `arm32` and `x86_32` architectures are not supported by NDK 25 via scons

Previously I used `Android.mk` to build with NDK 25 and it wasn't a problem. Now I found out that I can use `scons` to build for android directly to `.so`.
I simplified my CI script, but it turned out that 32-bit builds don't work with NDK 25.

Here are logs for arm32 with NDK 25: https://github.com/DmitriySalnikov/godot_qoi/actions/runs/3554264901/jobs/5970254673

And here are logs after changing version to NDK 23: https://github.com/DmitriySalnikov/godot_qoi/actions/runs/3554456032/jobs/5970566405

A copy of logs (in case CI logs are erased):
```
/usr/local/lib/android/sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ -o bin/libgodot_qoi.android.template_debug.arm32.so --target=armv7a-linux-androideabi18 -march=armv7-a -shared obj/qoi_import.os obj/qoi_save.os obj/qoi_wrapper.os obj/register_types.os -Lgodot-cpp/bin -lgodot-cpp.android.template_debug.arm32
ld: error: cannot open crtbegin_so.o: No such file or directory
ld: error: unable to find library -lc++
ld: error: cannot open crtend_so.o: No such file or directory
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [bin/libgodot_qoi.android.template_debug.arm32.so] Error 1
scons: building terminated because of errors.
Error: Process completed with exit code 2.
```

--- a/tools/android.py
+++ b/tools/android.py
@@ -8,7 +8,7 @@ def options(opts):
     opts.Add(
         "android_api_level",
         "Target Android API level",
-        "18" if "32" in ARGUMENTS.get("arch", "arm64") else "21",
+        "21",
     )
     opts.Add(
         "ANDROID_HOME",
@@ -47,11 +47,9 @@ def generate(env):
         my_spawn.configure(env)
 
     # Validate API level
-    api_level = int(env["android_api_level"])
-    if "64" in env["arch"] and api_level < 21:
-        print("WARN: 64-bit Android architectures require an API level of at least 21; setting android_api_level=21")
+    if int(env["android_api_level"]) < 21:
+        print("WARNING: minimum supported Android target api is 21. Forcing target api 21.")
         env["android_api_level"] = "21"
-        api_level = 21
 
     # Setup toolchain
     toolchain = get_android_ndk_root(env) + "/toolchains/llvm/prebuilt/"

